addons/forum-id/userscript.js

```diff
@@ -8,10 +8,41 @@ export default async function ({ addon, console, msg }) {
     addBtn.appendChild(addBtnAElement);
     addBtn.addEventListener("click", (e) => setTimeout(() => addIDLink(e), 0));
     addon.tab.appendToSharedSpace({ space: "forumsBeforePostReport", element: addBtn, scope: elm, order: 10 });
+    addon.tab.displayNoneWhileDisabled(addBtn, { display: "inline" });
   });
   function addIDLink(e) {
     let idName = e.target.closest(".blockpost").querySelector(".box-head > .conr").textContent;
     let id = e.target.closest(".blockpost").id.substring(1);
-    window.paste(`[url=https://scratch.mit.edu/discuss/post/${id}/]${idName}[/url] `);
+    window.paste(getIDLink(id, idName, true));
   }
+  function getIDLink(id, name, addSpace) {
+    return `[url=https://scratch.mit.edu/discuss/post/${id}/]${name}[/url]${addSpace ? " " : ""}`;
+  }
+  //Auto-adding IDs to quotes
+  const originalCopyPaste = window.copy_paste;
+  window.copy_paste = function (id) {
+    if (addon.self.disabled || !addon.settings.get("auto_add")) {
+      originalCopyPaste(id);
+      return;
+    }
+    var post = $("#" + id);
+    var username = post.find(".username").text();
+    $.ajax("/discuss/post/" + id.substr(1) + "/source/").done(function (data) {
+      paste(
+        "[quote=" +
+          username +
+          "][small](" +
+          getIDLink(
+            id.substring(1),
+            addon.settings.get("auto_add_use_number")
+              ? post["0"].querySelector(".box-head > .conr").textContent
+              : "View post",
```

Comment:
"View post" would need to be localized here (kind of - the forums themselves aren't localized completely).